src/ucp/tag/rndv.c

```diff
@@ -117,8 +117,10 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_rtr, (self),
     return status;
 }
 
-void ucp_tag_send_start_rndv(ucp_request_t *sreq)
+ucs_status_t ucp_proto_start_rndv(uct_pending_req_t *self)
 {
+    ucp_request_t *sreq = ucs_container_of(self, ucp_request_t, send.uct);
```

Comment:
I don't think these couple of instructions may affect somehow performance of RNDV in case of pended request but this proposal is aligned with design and adds flexibility for protocol implementation.